zum-back-end/api/index.js

```diff
@@ -0,0 +1,157 @@
+const fs = require("fs");
+const crypto = require("crypto");
+const puppeteer = require("puppeteer");
+const express = require("express");
+const router = express.Router();
+
+/* 랭킹 데이터 */
+router.get("/best", (req, res, next) => {
+  try {
+    let data = fs.readFileSync("cache/json/best.json");
+    data = JSON.parse(data);
+    res.json(data);
+  } catch (err) {
+    res.status(500).send(`Throw Exception -> ${err}`);
+  }
+});
+
+/* 카테고리별 데이터 */
+router.get("/content/:category", (req, res, next) => {
+  try {
+    // Data Check for Security
+    switch (req.params.category) {
+      case "life":
+      case "food":
+      case "culture":
+      case "travel":
+        break;
+      default:
+        throw new Error("Violating security rules");
+    }
+
+    // Data Size
+    const limit = req.query.limit !== undefined ? Number(req.query.limit) : 0;
+    const offset =
+      req.query.offset !== undefined ? Number(req.query.offset) : 12;
+
+    let data = fs.readFileSync(`cache/json/${req.params.category}.json`);
+    data = JSON.parse(data).slice(limit * offset, limit * offset + offset);
+    res.json(data);
+  } catch (err) {
+    res.status(500).send(`Throw Exception -> ${err}`);
+  }
+});
+
+/* 상세 페이지 데이터 */
+router.get("/content/:category/:idx", (req, res, next) => {
+  try {
+    // Data Check for Security
+    switch (req.params.category) {
+      case "life":
+      case "food":
+      case "culture":
+      case "travel":
+        break;
+      default:
+        throw new Error("Violating security rules");
+    }
+    if (!Number.isInteger(Number(req.params.idx))) {
+      throw new Error("Violating security rules");
+    }
+    const idx = Number(req.params.idx);
+
+    // categoryFindByIDX
+    let data = fs.readFileSync(`cache/json/${req.params.category}.json`);
+    data = JSON.parse(data);
+
+    // quick sort
+    data.sort((a, b) => {
+      return b.idx - a.idx;
+    });
+
+    //binary Search
+    let max = data.length - 1;
+    let half = Math.floor(data.length / 2);
+    let min = 0;
+    while (true) {
+      if (data[half].idx === idx) {
+        res.json(data[half]);
+        return;
+      } else if (half === min || half === max) {
+        if (data[max].idx === idx) res.json(data[half]);
+        else if (data[min].idx === idx) res.json(data[half]);
+        else res.json();
+        return;
+      } else if (data[half].idx < idx) {
+        // upper
+        max = half;
+        half -= Math.floor(Math.abs(max - min) / 2);
+      } else if (data[half].idx > idx) {
+        // under
+        min = half;
+        half += Math.floor(Math.abs(max - min) / 2);
+      }
+    }
```

Comment:
```suggestion
    res.json(data.find(v => v.idx === idx));
```
이렇게 한 줄로 처리할 수 있지 않을까요?